web-client/src/app/views/pin-reset/pin-reset.page.ts

```diff
@@ -41,45 +41,47 @@ export class PinResetPage implements OnInit {
 
   async onSubmit(answers: Map<string, string>) {
     this.isBusySaving = true;
-    const wallet_id: string = this.wallet_id ?? '';
-    const newSession = StartSgxSession.new(wallet_id);
-    const client_pk = newSession.our_public_key();
+    const wallet_id = this.wallet_id;
+    if (wallet_id) {
+      const newSession = StartSgxSession.new(wallet_id);
+      const client_pk = newSession.our_public_key();
 
-    const initialResult = await this.checkAnswers(
-      wallet_id,
-      answers,
-      client_pk
-    );
+      const initialResult = await this.checkAnswers(
+        wallet_id,
+        answers,
+        client_pk
+      );
 
-    if ('InvalidAuth' in initialResult) {
-      this.notification.showIncorrectSecutiyAnswers();
-    } else if ('Success' in initialResult) {
-      try {
-        const password = await this.newPin();
+      if ('InvalidAuth' in initialResult) {
+        this.notification.showIncorrectSecutiyAnswers();
+      } else if ('Success' in initialResult) {
+        try {
+          const password = await this.newPin();
 
-        if (password) {
-          const pinResetResult = await this.resetPin(
-            wallet_id,
-            password,
-            answers
-          );
+          if (password) {
+            const pinResetResult = await this.resetPin(
+              wallet_id,
+              password,
+              answers
+            );
 
-          if ('Reset' in pinResetResult) {
-            this.notification.showPinResetSuccessNotification();
-            this.navCtrl.navigateRoot('/');
-          } else {
-            this.notification.showUnexpectedFailureWarning();
-            this.navCtrl.navigateRoot('/');
+            if ('Reset' in pinResetResult) {
+              this.notification.showPinResetSuccessNotification();
+              this.navCtrl.navigateRoot('/');
+            } else {
+              this.notification.showUnexpectedFailureWarning();
+              this.navCtrl.navigateRoot('/');
+            }
           }
+        } catch (err) {
+          console.log('catch err: ' + err);
```

Comment:
Do we do nothing on error?

Reply:
Apart from logging the error, not really. Perhaps adding a popup notification displaying "An unexpected error".